board: merge phycore_am62x_a53 into phyboard_lyra_am62x

dnltz opened the pull request:
PHYTEC normally provides software for our kits like the phyBOARD-Lyra AM62x, which is our kit for the phyCORE-AM62x System on Module. Therefore, move the phycore_am62x_a53 board into phyboard_lyra_am62x and rename it.

Since I already touched the documentation, I also fixed some minor issues in the documentation and updated links to our latest software.

Added myself as collaborator to the TI K3 platform.

dnltz committed:
MAINTAINERS: Add myself as collaborator to TI K3

PHYTEC has support for the TI AM62x M4/A53 cores and we
plan to extend this support further in the future.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
boards: phytec: Merge AM62 A53 into phyboard_lyra_am62x

phyBOARD-Lyra AM62x is the official name for our kit, which we
use to develop our reference software. This kit includes the
phyCORE-AM62x with TI's AM62x SoC.

Rename the phycore_am62x_am6234_a53 board to phyboard_lyra_am62x_am6234_a53
and move all code into the phyboard_lyra_am62x directory. Thereby, the
A53 and M4 boards are located in the same directory and both named
correctly.

This is also a good chance to update documentation for both cores and
use links to our latest BSPs as well as our official AM62x Yocto machine
name for the AM62x HS-FS SOM.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

gramsay0 commented on boards/phytec/phyboard_lyra_am62x/phyboard_lyra_am62x_am6234_a53.dts:
```diff
@@ -9,8 +9,8 @@
 / {
-	model = "PHYTEC phyCORE-AM62x A53";
-	compatible = "ti,am6234";
+	model = "PHYTEC AM62x A54 phyBOARD-Lyra";
```
typo: ~A54~ A53

gramsay0 commented:
Rename the deprecated board name:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/deprecated.cmake#L633
I assume you don't need to deprecate `phycore_am62x/am6234/a53` since it has not been in a release yet

gramsay0 commented on boards/phytec/phyboard_lyra_am62x/doc/phyboard_lyra_am62x_a53.rst:
```diff
@@ -78,7 +83,7 @@ You can build an application in the usual way. Refer to
 
 .. zephyr-app-commands::
-   :board: phycore_am62x/am6234/a53
+   :board: phyboard_am62x/am6234/a53
```
Should this be `phyboard_lyra_am62x/am6234/a53`?

gramsay0 reviewed (commented):
I'm not really up to speed with the HW v2 naming, but would changing the board names `phyboard_lyra_am62x/am6234/a53` to `phyboard_lyra/am6234/a53` make sense? At first glance the `am62x` seem unnecessary

dnltz committed:
boards: phytec: Remove am62x from phyboard_lyra_am62x

Both phyBOARD-Lyra AM62x boards were called
phyboard_lyra_am62x/am6234/a53 or phyboard_lyra_am62x/am6234/m4.
Since the AM62x SOC part number is now included in the board name
as am6234, there is no need to have am62x in the board name itself.

Moreover, PHYTEC also has an AM62Ax SOM which comes with the same
carrier-board. Integration of this kit would be easier later.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz committed:
boards: deprecated: Update PHYTEC phyBOARD-Lyra AM62x

The PHYTEC phyboard_lyra_am62x board was renamed to phyboard_lyra
to simplify the board name.

The PHYTEC phycore_am62x board was renamed and moved into
phyboard_lyra as phyboard_lyra/am6234/a53.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

dnltz commented:
> I'm not really up to speed with the HW v2 naming, but would changing the board names `phyboard_lyra_am62x/am6234/a53` to `phyboard_lyra/am6234/a53` make sense? At first glance the `am62x` seem unnecessary

you're right. The AM62x part number is now part of the board and therefore not required in the phyboard_lyra name anymore. We also have the AM62Ax which comes with the same carrier board.

Technically, it could be called am62x4 since the third digit defines the present of a GPU and display subsystem, but should be fine for now.

dnltz committed:
MAINTAINERS: Add boards/phytec/phyboard_lyra to TI K3

We dropped the am62x from the phyboard_lyra_am62x name. Afterwards,
the files section for the TI K3 Platforms doesn't fit anymore.
Add the phyboard_lyra directly as all kits for this carrier-boards
use a K3 SOC.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>

nordicjm commented on boards/phytec/phyboard_lyra_am62x/doc/phyboard_lyra_am62x_a53.rst:
```diff
@@ -20,13 +20,18 @@ Zephyr OS is ported to run on the Cortex®-A53 core.
 ==================
 
-The Zephyr phycore_am62x_a53 board configuration supports the following hardware
+The Zephyr phyboard_lyra_am62x_am6234_a53 board configuration supports the following hardware
```
slashes

nordicjm commented on boards/phytec/phyboard_lyra_am62x/doc/phyboard_lyra_am62x_m4.rst:
```diff
@@ -43,7 +43,7 @@ specifications are used:
 ==================
 
-The am62x_m4_phyboard_lyra configuration supports the following hardware features:
+The phyboard_lyra_am62x_am6234_m4 configuration supports the following hardware features:
```
slashes

nordicjm replied:
Fix the issues in the first commit

dnltz replied:
You reviewed an outputed file again.

nordicjm commented on boards/phytec/phyboard_lyra/doc/phyboard_lyra_am62xx_m4.rst:
```diff
@@ -43,7 +43,7 @@ specifications are used:
 ==================
 
-The phyboard_lyra_am62x_am6234_m4 configuration supports the following hardware features:
+The phyboard_lyra_am6234_m4 configuration supports the following hardware features:
```
slashes

gramsay0 commented on MAINTAINERS.yml:
```diff
@@ -3633,8 +3633,9 @@ TI K3 Platforms:
     - vaishnavachath
   collaborators:
     - gramsay0
+    - dnltz
```
@dnltz are you a Zephyr "contributor"?
(I just discovered this doesn't do anything otherwise)

dnltz replied:
I would say yes. According to https://docs.zephyrproject.org/latest/project/project_roles.html, it's anyone who contributes which I already did before.

gramsay0 replied:
I thought that too, but you have to join the Zephyr github project first.
You can apply by making a "Contributor Nomination" issue:
https://github.com/zephyrproject-rtos/zephyr/issues/new/choose

dnltz replied:
interesting ... thanks! Opened an issue to nominate myself

nordicjm commented on boards/phytec/phyboard_lyra_am62x/doc/phyboard_lyra_am62x_a53.rst:
```diff
@@ -20,13 +20,18 @@ Zephyr OS is ported to run on the Cortex®-A53 core.
 ==================
 
-The Zephyr phycore_am62x_a53 board configuration supports the following hardware
+The Zephyr phyboard_lyra_am62x_am6234_a53 board configuration supports the following hardware
```
use proper name with slashes

nordicjm replied:
Did you even check your own PR? https://github.com/zephyrproject-rtos/zephyr/pull/70913/commits/477835129407fa80dcd50005209b61f295054213#diff-bb3ed069bee91ffe0d661beedb71369e2c7c1947913136ed4134428b8f51e69eR34 this is literally your second commit in the current version of your PR

dnltz replied:
I know what you mean. I just don't see any reason in cleaning up the history for documentation only, while the output is absolutely fine. No one will benefit from that change... the only thing that will happen is that 3 people have to review and approve again, which is IMO a waste of time for these people.

nordicjm commented on boards/phytec/phyboard_lyra/Kconfig.phyboard_lyra:
```diff
@@ -5,8 +5,8 @@
-config BOARD_PHYBOARD_LYRA_AM62X
-	select SOC_AM6234_A53 if BOARD_PHYBOARD_LYRA_AM62X_AM6234_A53
-	select SOC_AM6234_M4 if BOARD_PHYBOARD_LYRA_AM62X_AM6234_M4
+config BOARD_PHYBOARD_LYRA
+	select SOC_AM6234_A53 if BOARD_PHYBOARD_LYRA_AM6234_A53
+	select SOC_AM6234_M4 if BOARD_PHYBOARD_LYRA_AM6234_M4
```
make changes in the first commit

dnltz replied:
Commit title is `boards: phytec: Remove am62x from phyboard_lyra_am62x`. Why should I move this change into another commit?

nordicjm replied:
You're adding hwmv1 stuff which was outdated a month ago then correcting it in another commit, this is a flat out no

dnltz replied:
This change just drops am62x. How is this related to hwmv1?

nordicjm replied:
https://github.com/zephyrproject-rtos/zephyr/pull/70913/commits/477835129407fa80dcd50005209b61f295054213 this change, then in the next commit doing
```
- The Zephyr phyboard_lyra_am62x_am6234_a53 board configuration supports the following hardware
+ The Zephyr phyboard_lyra/am6234/a53 board configuration supports the following hardware
```
`phyboard_lyra_am62x_am6234_a53` is hwmv1, let's stick to hwmv2 throughout rather than having commits adding hwmv1 terminology

dnltz replied:
isn't this thread about something different? Pushed a clean history btw.

dnltz commented:
@stephanosio gentle ping

nordicjm reviewed (changes requested):
You keep marking comments as resolved and yet they're not resolved
